conda-store-server/conda_store_server/alembic/versions/46bdf428642d_change_namespacerolemapping_and_.py

```diff
@@ -0,0 +1,47 @@
+"""Change NamespaceRoleMapping and Namespace
+
+Revision ID: 46bdf428642d
+Revises: b387747ca9b7
+Create Date: 2023-10-08 10:40:06.227854
+
+"""
+import sqlalchemy as sa
+from alembic import op
+
+# revision identifiers, used by Alembic.
+revision = "46bdf428642d"
+down_revision = "b387747ca9b7"
+branch_labels = None
+depends_on = None
+
+
+def upgrade():
+    op.create_table(
+        'namespace_role_mapping_new',
+        sa.Column('id', sa.Integer(), nullable=False),
+        sa.Column('namespace_id', sa.Integer(), nullable=False),
+        sa.Column('other_namespace_id', sa.Integer(), nullable=False),
+        sa.Column('role', sa.Unicode(length=255), nullable=False),
+        sa.ForeignKeyConstraint(['namespace_id'], ['namespace.id'], ),
+        sa.ForeignKeyConstraint(['other_namespace_id'], ['namespace.id'], ),
+        sa.PrimaryKeyConstraint('id'),
+        sa.UniqueConstraint('namespace_id', 'other_namespace_id', 'role', name='_uc')
+    )
+    # Note: data is NOT copied before dropping the old table
+    op.drop_table("namespace_role_mapping")
+    op.rename_table("namespace_role_mapping_new", "namespace_role_mapping")
+
+
+def downgrade():
+    op.create_table(
+        'namespace_role_mapping_new',
+        sa.Column('id', sa.Integer(), nullable=False),
+        sa.Column('namespace_id', sa.Integer(), nullable=False),
+        sa.Column('entity', sa.Unicode(length=255), nullable=False),
+        sa.Column('role', sa.Unicode(length=255), nullable=False),
+        sa.ForeignKeyConstraint(['namespace_id'], ['namespace.id'], ),
+        sa.PrimaryKeyConstraint('id')
+    )
+    # Note: data is NOT copied before dropping the old table
+    op.drop_table("namespace_role_mapping")
+    op.rename_table("namespace_role_mapping_new", "namespace_role_mapping")
```

Comment:
Migration seems to work with Postgres and SQLite. But it breaks new API tests, which used to pass, probably because I'm not copying data, so namespace URLs 404:

```
FAILED tests/test_api.py::test_update_namespace_metadata[True] - requests.exceptions.HTTPError: 404 Client Error: Not Found for url: http://localhost:5000/conda-store/api/v1/namespace/...
FAILED tests/test_api.py::test_update_namespace_metadata[False] - assert 404 == 403
FAILED tests/test_api.py::test_crud_namespace_roles[True] - requests.exceptions.HTTPError: 404 Client Error: Not Found for url: http://localhost:5000/conda-store/api/v1/namespace/...
FAILED tests/test_api.py::test_crud_namespace_roles[False] - assert 404 == 403
```

Reply:
Hmm, it passed on CI. But `test_namespace_role_mapping` needs to be updated to use new schema anyway.